tests/vm_startup_test.go

```diff
@@ -0,0 +1,162 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+)
+
+const timeout = 3 * time.Minute
+const pollingInterval = 5 * time.Second
+const testNamespace = "secondary-test"
+const dnsPort = "31111"
+const dnsIP = "127.0.0.1" // Forwarded to the node port - https://github.com/kubevirt/kubevirtci/pull/867
+const domain = "vm.secondary.io"
+
+var _ = Describe("Virtual Machines Startup", func() {
+
+	BeforeEach(func() {
+		By("Creating test namespace")
+		ns := k8sv1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testNamespace,
+			},
+		}
+		err := getClient().Create(context.Background(), &ns)
+		Expect(err).ToNot(HaveOccurred(), "couldn't create namespace for the test")
+	})
+
+	AfterEach(func() {
+		By("Removing all Virtual Machine Instances from the namespace")
+		vmiList := &v1.VirtualMachineInstanceList{}
+		err := getClient().List(context.Background(), vmiList, &client.ListOptions{Namespace: testNamespace})
+		Expect(err).ToNot(HaveOccurred())
+
+		for _, vmiObject := range vmiList.Items {
+			err = getClient().Delete(context.Background(), &vmiObject)
+			Expect(err).ToNot(HaveOccurred())
+		}
```

Comment:
isn't there a way to just delete in one shot all vms from a given namespace using the client ?
(if so, same for nads etc)

Note that the Expect in the for means that if an error occur, it will skip the rest, 
i guess it is fine for now, maybe in the future we would prefer to collect errors but continue and assert the errors at the end